src/stdlib_sparse_conversion.fypp

```diff
@@ -0,0 +1,452 @@
+#:include "common.fypp"
+#:set R_KINDS_TYPES = list(zip(REAL_KINDS, REAL_TYPES, REAL_SUFFIX))
+#:set C_KINDS_TYPES = list(zip(CMPLX_KINDS, CMPLX_TYPES, CMPLX_SUFFIX))
+#:set KINDS_TYPES = R_KINDS_TYPES+C_KINDS_TYPES
+!> The `stdlib_sparse_conversion` module provides sparse to sparse matrix conversion utilities.
+!>
+! This code was modified from https://github.com/jalvesz/FSPARSE by its author: Alves Jose
+module stdlib_sparse_conversion
+    use stdlib_kinds, only: int8, int16, int32, int64, sp, dp, xdp, qp
+    use stdlib_sparse_kinds
+    implicit none
+    private
+
+    public :: dense2coo, coo2dense, coo2csr, csr2coo
+    public :: csr2sellc
+    public :: coo2ordered
+
+    !> Conversion from dense to coo
+    !> 
+    !> Enables extracting the non-zero elements of a dense 2D matrix and
+    !> storing those values in a COO format. The coo matrix is (re)allocated on the fly.
+    interface dense2coo
+        #:for k1, t1, s1 in (KINDS_TYPES)
+        module procedure dense2coo_${s1}$
+        #:endfor
+    end interface
+
+    !> Conversion from coo to dense
+    !> 
+    !> Enables creating a dense 2D matrix from the non-zero values tored in a COO format
+    !> The dense matrix can be allocated on the fly if not pre-allocated by the user.
+    interface coo2dense
+        #:for k1, t1, s1 in (KINDS_TYPES)
+        module procedure coo2dense_${s1}$
+        #:endfor
+    end interface
+
+    !> Conversion from coo to csr
+    !> 
+    !> Enables transferring data from a COO matrix to a CSR matrix
+    !> under the hypothesis that the COO is already ordered.
```

Comment:
What are the implications of an unordered COO matrix? Would that break any other manipulation routines? 
If so, I would recommend to always run a sort internally whenever these operations take place (it would be even better to keep track of it via an `is_sorted` flag inside the derived type, if it is not too much work). 

In other words, I would suggest to go for safety first, and performance later. Another way to do this may be to use an optional flag, that assumes *not* sorted by default: 

```Fortran
subroutine coo2dense(COO, is_sorted=.false.) 
```